hypervisor/mmu.c

```diff
@@ -0,0 +1,31 @@
+#include "common.h"
+#define MAX_PAGE_PDE    8
+unsigned int page_directory[1024] __attribute__((aligned(4096)));
+unsigned int page_tables[MAX_PAGE_PDE][1024] __attribute__((aligned(4096)));
+
+
+void init_page_directory() {
+    int i;
+    int j;
+    for(i = 0; i < MAX_PAGE_PDE; i++)
+    {
+        // This sets the following flags to the pages:
+        //   Supervisor: Only kernel-mode can access them
+        //   Write Enabled: It can be both read from and written to
+        //   Not Present: The page table is not present
+        page_directory[i] = 0x00000002;
+    }
+
+    for (j=0; j<MAX_PAGE_PDE; j++) {
+        for(i = 0; i < 1024; i++)
+        {
+            // As the address is page aligned, it will always leave 12 bits zeroed.
+            // Those bits are used by the attributes ;)
+            page_tables[j][i] = ((i * 0x1000) + (j * 0x400000)) | 3; // attributes: supervisor level, read/write, present.
```

Comment:
do not directly using 0x1000, 0x400000

Reply:
we have to directly use those values. 
because It is not only a value, 0x1000, but also means low bits are 0, bit12 is 1.

0x400000 is the mapping size of one of entrys of page_directory.

Reply:
0x1000 means one page can reach 4KB
0x400000 means one page dir can reach 4MB  
so both are the constant. correct me if i am misunderstanding.